zebra-state/src/service/finalized_state/disk_format/upgrade.rs

```diff
@@ -62,39 +89,39 @@ pub enum DbFormatChange {
 /// Cancelling the thread on drop has a race condition, because two handles can be dropped at
 /// the same time.
 ///
-/// If cancelling the thread is important, the owner of the handle must call force_cancel().
+/// If cancelling the thread is required for correct operation or usability, the owner of the
+/// handle must call force_cancel().
 #[derive(Clone, Debug)]
 pub struct DbFormatChangeThreadHandle {
-    /// A handle that can wait for the running format change thread to finish.
+    /// A handle to the format change/check thread.
+    /// This thread continues running so it can perform periodic format checks.
     ///
     /// Panics from this thread are propagated into Zebra's state service.
     /// The task returns an error if the upgrade was cancelled by a shutdown.
-    update_task: Option<Arc<JoinHandle<Result<(), CancelFormatChange>>>>,
+    update_task: Option<Arc<JoinHandle<Result<Infallible, CancelFormatChange>>>>,
 
-    /// A channel that tells the running format thread to finish early.
     /// A channel that tells the running format thread to finish early.
     cancel_handle: mpsc::SyncSender<CancelFormatChange>,
 }
 
-/// Marker for cancelling a format upgrade.
+/// Marker type that is sent to cancel a format upgrade, and returned as a error on cancellation.
```

Comment:
###### Optional Suggestion
```suggestion
/// Marker type that is sent to cancel a format upgrade, and returned as an error on cancellation.
```

Reply:
I'm going to fix this in the next PR instead.